app/services/SubmitService.scala

```diff
@@ -36,10 +37,16 @@ class SubmitService @Inject()(
                   (implicit ec: ExecutionContext, headerCarrier: HeaderCarrier): Future[Result] = {
     ua.get(VersionInfoPage) match {
       case Some(VersionInfo(version, Compiled)) =>
-        eventReportingConnector.submitReport(pstr, ua, version.toString).flatMap { _ =>
-          val updatedUA = ua.setOrException(VersionInfoPage, VersionInfo(version, Submitted), nonEventTypeData = true)
-          userAnswersCacheConnector.save(pstr, updatedUA).map { _ => Ok }
+        eventReportingConnector.submitReport(pstr, ua, version.toString).flatMap { response =>
```

Comment:
eventReportingConnector.submitReport should return Future[HttpResponse] and this function also should return Future[HttpResponse] and should be handled via controller.